qemu/tests/boot_from_nbd_image.py

```diff
@@ -0,0 +1,57 @@
+import socket
+
+from virttest import qemu_storage
+
+from avocado import fail_on
+
+from avocado.utils import process
+
+from provider import qemu_img_utils as img_utils
+
+from provider.nbd_image_export import QemuNBDExportImage
+
+
+def run(test, params, env):
+    """
+    1) Create a local raw/qcow2/luks image
+    2) Export it with qemu-nbd
+    3) Convert remote system image to the exported nbd image
+    4) Start VM from the exported image
+    5) Log into VM
+    """
+    def _convert_image():
+        source = params['images'].split()[0]
+        target = params['convert_target']
+        source_params = params.object_params(source)
+        target_params = params.object_params(target)
+        source_image = qemu_storage.QemuImg(source_params, None, source)
+
+        # Convert source to target
+        fail_on((process.CmdError,))(source_image.convert)(
+            target_params, None, skip_target_creation=True)
+
+    nbd_export = QemuNBDExportImage(params["local_image_tag"], params)
+    nbd_export.create_image()
+    nbd_export.export_image()
+
+    # we only export image with local nbd server
+    localhost = socket.gethostname()
+    params['nbd_server_%s' % params['convert_target']
+           ] = localhost if localhost else 'localhost'
+
+    session = None
+    vm = None
+    try:
+        _convert_image()
+        vm = img_utils.boot_vm_with_images(test, params, env,
+                                           (params['convert_target'],))
+        session = vm.wait_for_login(
+            timeout=int(params.get("login_timeout", 480)))
+        if not session:
+            test.fail('Failed to log into VM')
+    finally:
+        if session:
+            session.close()
+        if vm:
+            vm.destroy()
+        nbd_export.stop_export()
```

Comment:
The order `fianlly` must be `session.close()` >` vm.destroy() `-> `nbd_export.stop_export()` ?
if just only `nbd_export.stop_export()` , the will be error?
if not, I think we could not `session = None` and   `  vm = None` then
e.g:
```
try
   _convert_image()
   vm = img_utils.boot_vm_with_image
   session = vm.wait_for_log
finally:
   nbd_export.stop_export()
```

Reply:
it's better to close session, then shutdown vm, and finally stop export, this is the regular process, for the exported nbd image is vm's system image.